FWCore/Framework/interface/PathsAndConsumesOfModules.h

```diff
@@ -62,12 +72,17 @@ namespace edm {
     // following data member
     std::vector<std::pair<unsigned int, unsigned int> > moduleIDToIndex_;
 
-    std::vector<std::vector<ModuleDescription const*> > modulesWhoseProductsAreConsumedBy_;
+    std::vector<std::vector<ModuleDescription const*> > modulesWhoseProductsAreConsumedByEvent_;
+    std::vector<std::vector<ModuleDescription const*> > modulesWhoseProductsAreConsumedByLumiRun_;
+    std::vector<std::set<ModuleProcessName> > modulesInPreviousProcessesWhoseProductsAreConsumedBy_;
```

Comment:
Does this really need to be a `set` instead of a sorted `vector`? What are the different entries in the `vector`?

Reply:
> Does this really need to be a `set` instead of a sorted `vector`? 

It can be a sorted `vector`.

> What are the different entries in the `vector`?

The (outermost) `vector` has an entry for each module in the (Sub)Process.

Reply:
Changed the `set` to sorted `vector`.